fatiando/gridder.py

```diff
@@ -439,3 +445,325 @@ def cut(x, y, scalars, area):
+    elif padtype.lower() == 'oddreflection':
+        # Pad with odd reflection
+        ap = numpy.pad(a, nps, mode='reflect',
+                       reflect_type='odd')
+    elif padtype.lower() == 'oddreflectiontaper':
+        # Pad with odd reflection and a cosine taper to mean
+        ap = (numpy.pad(a, nps, mode='reflect',
+                        reflect_type='odd') - m)
+        for ii in range(0, nd):
+            ap = numpy.apply_along_axis(_costaper, ii, ap, lp=nps[ii][0],
+                                        rp=nps[ii][1])
+        ap += m
+
+    return ap, nps
+
+
+def unpad_array(a, nps):
+    """
+    Unpads an array using the outputs from pad_array.
+
+    This function takes a padded array and removes the padding from both.
+    Effectively, this is a complement to gridder.cut for when you already
+    know the number of elements to remove.
+
+    .. note:: Unlike :func:`~fatiando.gridder.pad_array`, this returns a slice
+        of the input array.  Therefore, any changes to the padded array will be
+        reflected in the unpadded array.
+
+    Parameters:
+
+    * a : array
+        Array to be un-padded.  Can be of arbitrary dimension.
+    * nps : list
+        List of tuples giving the min and max indices for the cutoff.
+        Identical to nps returned by pad_array
+
+    Returns:
+
+    * b : array
+        Array of same dimension as a, with padding removed
+
+    Examples:
+
+        >>> z = numpy.array([3, 4, 4, 5, 6])
+        >>> zpad, nps = pad_array(z)
+        >>> zpad
+        array([ 4.4,  3.2,  3. ,  4. ,  4. ,  5. ,  6. ,  4.4])
+        >>> zunpad = unpad_array(zpad, nps)
+        >>> zunpad
+        array([ 3.,  4.,  4.,  5.,  6.])
+
+    """
+    o = []
+    for ii in range(0, a.ndim):
+        o.append(slice(nps[ii][0], a.shape[ii] - nps[ii][1]))
+    b = a[o]
+
+    return b
+
+
+def pad_coords(xy, shape, nps):
+    """
+    Pads coordinate vectors.
+
+    Designed to be used in concert with :func:`~fatiando.gridder.pad_array`,
+    this function takes a list of coordinate vectors and pads them using the
+    same discretization.
+
+    Parameters:
+
+    * xy : list
+        List of arrays of coordinates
+    * shape : tuple
+        Size of original array
+    * nps : tuple
+        Size of padded array
+
+    Returns:
+
+    * coordspad : list
+        List of padded coordinate arrays
+
+    Examples:
+
+        >>> shape = (5, 6)
+        >>> x, y, z = regular((-10,10,-20,0), shape, z=-25)
+        >>> gz = numpy.zeros(shape)
+        >>> gzpad, nps = pad_array(gz)
+        >>> x.reshape(shape)[:,0]
+        array([-10.,  -5.,   0.,   5.,  10.])
+        >>> y.reshape(shape)[0,:]
+        array([-20., -16., -12.,  -8.,  -4.,   0.])
+        >>> xy = [x, y]
+        >>> [xp, yp] = pad_coords(xy, shape, nps)
+        >>> xp
+        array([-20., -15., -10.,  -5.,   0.,   5.,  10.,  15.])
+        >>> yp
+        array([-24., -20., -16., -12.,  -8.,  -4.,   0.,   4.])
```

Comment:
Does `pad_coords` always return the arrays with the `meshgrid` applied? If so, then we're leaving it up to the user to apply the `meshgrid`? This sounds like something that will generate a lot of emails on the mailing list :wink: Getting the order of arguments for meshgrid is very confusing so maybe it would be best to return the flattened arrays after meshgrid. That way the user can treat the new arrays just like the old ones.